src/Suave/Cookie.fs

```diff
@@ -42,211 +42,224 @@ module Cookie =
+  let refreshCookies relativeExpiry httpCookie : WebPart =
+    slidingExpiry relativeExpiry httpCookie ||> setPair
+
+  let updateCookies (csctx : CookiesState) f_plainText : WebPart =
+    context (fun ctx ->
+      let logger = ctx.runtime.logger
+      let plainText =
+        match readCookies csctx.serverKey csctx.cookieName ctx.response.cookies with
+        | Choice1Of2 (_, plainText) ->
+          Log.log logger "Suave.Cookie.updateCookies" LogLevel.Debug "updateCookies - existing"
+          f_plainText (Some plainText)
         | Choice2Of2 _ ->
-          Log.log logger "Suave.Cookie.update_cookies" LogLevel.Debug "update_cookies - first time"
-          f_plain_text None
+          Log.log logger "Suave.Cookie.updateCookies" LogLevel.Debug "updateCookies - first time"
+          f_plainText None
 
       /// Since the contents will completely change every write, we simply re-generate the cookie
-      generate_cookies csctx.server_key csctx.cookie_name
-                       csctx.relative_expiry csctx.secure
-                       plain_text'
-      ||> set_pair
-      >>= Writers.set_user_data csctx.user_state_key plain_text')
+      generateCookies csctx.serverKey csctx.cookieName
+                       csctx.relativeExpiry csctx.secure
+                       plainText
+      ||> setPair
+      >>= Writers.setUserData csctx.userStateKey plainText)
 
-  let cookie_state (csctx : CookiesState)
+  let cookieState (csctx : CookiesState)
                    // unit -> plain text to store OR something to run of your own!
-                   (no_cookie : unit -> Choice<byte [], WebPart>)
-                   (decryption_failure   : _ -> Choice<byte [], WebPart>)
+                   (noCookie : unit -> Choice<byte [], WebPart>)
+                   (decryptionFailure   : _ -> Choice<byte [], WebPart>)
                    (f_success : WebPart)
                    : WebPart =
     context (fun ({ runtime = { logger = logger }} as ctx) ->
 
       let log = Log.log logger "Suave.Cookie.cookie_state" LogLevel.Debug
 
-      let set_cookies plain_text =
-        let http_cookie, client_cookie =
-          generate_cookies csctx.server_key csctx.cookie_name
-                           csctx.relative_expiry csctx.secure
-                           plain_text
-        set_pair http_cookie client_cookie >>=
-          Writers.set_user_data csctx.user_state_key plain_text
+      let setCookies plainText =
+        let httpCookie, clientCookie =
+          generateCookies csctx.serverKey csctx.cookieName
+                           csctx.relativeExpiry csctx.secure
+                           plainText
+        setPair httpCookie clientCookie >>=
+          Writers.setUserData csctx.userStateKey plainText
 
-      match read_cookies csctx.server_key csctx.cookie_name (ctx.request |> HttpRequest.cookies) with
-      | Choice1Of2 (http_cookie, plain_text) ->
+      match readCookies csctx.serverKey csctx.cookieName ctx.request.cookies with
```

Comment:
Yes, that's a recurring problem with ".cookies", ".paramData" and ".formData", all of which parse.  The options are
 
1. move to use an object instead of a record to allow these to be cached internally in the object
1. add a cache field to the record to allow these to be cached internally in the record
1. use a property name like cookieData 
1. use a method like getCookieData

I think 1 or 2 is easiest for the user - no thinking is required. 1 is really the "right" way to do things but then you lose record syntax (see https://fslang.uservoice.com/forums/245727-f-language/suggestions/6420709-allow-record-like-obj-with-newvals-syntax-for).  So 2 may be the right option.

Alternatively revert to the function HttpRequest.cookies, but it's somehow a bit clumsy in the common case.